internal/pin/auto.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package pin
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/DataDog/orchestrion/internal/ensure"
+	"github.com/DataDog/orchestrion/internal/log"
+	"github.com/DataDog/orchestrion/internal/version"
+	"github.com/charmbracelet/lipgloss"
+	"golang.org/x/term"
+)
+
+const (
+	envVarCheckedGoMod = "DD_ORCHESTRION_IS_GOMOD_VERSION"
+	envValTrue         = "true"
+)
+
+var (
+	requiredVersionError error // Whether the go.mod version check succeeded
+)
+
+// AutoPinOrchestrion automatically runs `pinOrchestrion` if the necessary
+// requirements are not already met. It prints messages to `os.Stderr` to inform
+// the user about what is going on.
+func AutoPinOrchestrion() {
+	if requiredVersionError == nil {
+		// Nothing to do!
+		return
+	}
```

Comment:
I suspect that you have to rebase on main because this looks like a code I changed in #372